source/GameAction.cpp

```diff
@@ -182,6 +182,23 @@ void GameAction::LoadSingle(const DataNode &child)
 		else
 			child.PrintTrace("Error: Skipping invalid \"fine\" with non-positive value:");
 	}
+	else if(key == "mortgage" && hasValue)
+	{
+		GameAction::Mortgage mortgage;
+		mortgage.principal = max<int64_t>(0, child.Value(1));
+		for(const DataNode &grand : child)
+		{
+			const string &grandKey = grand.Token(0);
+			bool grandHasValue = (grand.Size() > 1);
+			if(grandKey == "term" && grandHasValue)
+				mortgage.term = max<int>(1, grand.Value(1));
+			else if(grandKey == "interest" && grandHasValue)
+				mortgage.interest = clamp(grand.Value(1), 0., 0.999);
+			else
+				grand.PrintTrace("Error: Skipping unrecognized \"mortgage\" attribute:");
+		}
+		mortgages.emplace_back(mortgage);
+	}
```

Comment:
Why not just add a new item to the vector and edit its members rather than making one locally, editing it, and then copying that into the vector?